lib/transports.js

```diff
@@ -1,14 +1,16 @@
 var events = require('events');
 var util = require('util');
+var assign = require('object-assign');
```

Comment:
> You worry about extra dependency?

Yes. I don't want to have to traverse into external code and debug an issue for something so trivial. For example, for me to even figure out what this code was doing, I had to look this up.

I first google'd "object-assign", and it took me to: https://www.npmjs.com/package/object-assign Which is correct, then I was like, "I still don't understand what's going on here." Which brought me to https://github.com/sindresorhus/object-assign, and it's described as the "ES6 Object.assign() polyfill". Which links me to http://www.2ality.com/2014/01/object-assign.html. Since this wasn't official documentation, I ended up googling Object.assign, which brought me to the MDN page: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign which finally says in plain English: `The Object.assign() method is used to copy the values of all enumerable own properties from one or more source objects to a target object. It will return the target object.`

:)

So yes, I'd prefer a simple 1 line for loop over this hunt trying to figure out what `assign` means personally.

It also may be because I'm old and don't know anything about ES6 so it wasn't immediately obvious what it did, and how it's different from what I proposed. I still don't actually know how it's much different except for the fact that it can merge multiple objects and we don't need that functionality.